pkg/utils/kubernetes/kubernetes.go

```diff
@@ -383,3 +399,147 @@ func MergeOwnerReferences(references []metav1.OwnerReference, newReferences ...m
 
 	return references
 }
+
+// OwnedBy checks if the given object's owner reference contain an entry with the provided attributes.
+func OwnedBy(obj runtime.Object, apiVersion, kind, name string, uid types.UID) bool {
+	acc, err := meta.Accessor(obj)
+	if err != nil {
+		return false
+	}
+
+	for _, ownerReference := range acc.GetOwnerReferences() {
+		return ownerReference.APIVersion == apiVersion &&
+			ownerReference.Kind == kind &&
+			ownerReference.Name == name &&
+			ownerReference.UID == uid
+	}
+
+	return false
+}
+
+// NewestObject returns the most recently created object based on the provided list object type. If a filter function
+// is provided then it will be applied for each object right after listing all objects. If no object remains then nil
+// is returned.
+func NewestObject(ctx context.Context, c client.Client, listObj runtime.Object, filterFn func(runtime.Object) bool, listOpts ...client.ListOption) (runtime.Object, error) {
+	if !meta.IsListType(listObj) {
+		return nil, fmt.Errorf("provided <listObj> is not a List type")
+	}
+
+	if err := c.List(ctx, listObj, listOpts...); err != nil {
+		return nil, err
+	}
+
+	if filterFn != nil {
+		var items []runtime.Object
+
+		if err := meta.EachListItem(listObj, func(obj runtime.Object) error {
+			if filterFn(obj) {
+				items = append(items, obj)
+			}
+			return nil
+		}); err != nil {
+			return nil, err
+		}
+
+		if err := meta.SetList(listObj, items); err != nil {
```

Comment:
Nit, deep copy list before mutating it further. Mutating the incoming object directly might be unexpected for the caller of this function.

Reply:
Hm, why? It's already being "mutated" because we run `List()` and store the results in `listObj`. 
Now, if the user provides a `filterFn` then we filter the objects accordingly and update `listObj`. This is exactly what I would expect as a user when I provide a filter function, similarly to how I expect to find all objects in `listObj` when I call `List()` only.

Reply:
> similarly to how I expect to find all objects in listObj when I call List() only.

The `List()` function explicitly says that it takes the `listObj` to do that.

In this case, you have a `NewestObject` function that is devoted to find the newest object only, but as a byproduct it also mutates the passed `listObj`. This is unexpected from my point of view.

Reply:
Let me add a doc string to clarify on this then. I explicitly want this behaviour so that the user can also run additional logic against the resulting items without the necessity of manually applying his filter function again to get rid of the undesired objects.

Reply:
Okay, thanks for clarifying. Can you also verify the `listObj` in the unit tests?

Reply:
👍🏻 done